plugins/languages/vimtex.nix

```diff
@@ -24,21 +24,56 @@ in
         '';
         default = {};
       };
+
+      viewMethod = mkOption {
+        type = types.str;
+        description = ''
+          The view method that vimtex will use to display PDF's.
+          Check https://github.com/lervag/vimtex/blob/03c83443108a6984bf90100f6d00ec270b84a339/doc/vimtex.txt#L3322
+          for more information.
+        '';
+        default = "general";
+      };
+      installTexLive = mkOption {
+        type = types.bool;
+        description = ''
+                 Whether or not to install TexLive.
+          See https://nixos.wiki/wiki/TexLive.
+        '';
+        default = false;
+      };
+      texLivePackage = helpers.mkPackageOption "texLivePackage" pkgs.texlive.combined.scheme-medium;
     };
 
     config = let
       globals =
         {
           enabled = cfg.enable;
           callback_progpath = "nvim";
+          view_method = cfg.viewMethod;
         }
         // cfg.extraConfig;
+      basePackages =
+        [pkgs.pstree]
+        ++ (
+          if cfg.installTexLive
+          then [cfg.texLivePackage]
+          else []
+        );
+      viewMethodAndPDFViewerPairs = {
+        general = basePackages ++ [pkgs.xdotool];
+        zathura = basepackages ++ [pkgs.xdotool pkgs.zathura];
+        zathura_simple = basePackages ++ [pkgs.zathura];
+        mupdf = basepackages ++ [pkgs.xdotool pkgs.mupdf];
+      };
     in
       mkIf cfg.enable {
         extraPlugins = [cfg.package];
 
-        # Usefull for inverse search
-        extraPackages = with pkgs; [pstree xdotool];
+        extraPackages =
+          if hasAttr "${cfg.viewMethod}" viewMethodAndPDFViewerPairs
+          then viewMethodAndPDFViewerPairs."${cfg.viewMethod}"
+          else basePackages;
```

Comment:
And here doing `basePackages ++ (lib.optionals (hasAttr "${cfg.viewMethod}" viewMethodAndPDFViewerPairs) viewMethodAndPDFViewerPairs.${cfg.viewMethod})`

Reply:
Fixed as well

Reply:
Since it is now `basepackages ++` I have removed it from `viewMethodAndPDFViewerPairs`